skymp5-server/cpp/server_guest_lib/script_classes/PapyrusLeveledItem.cpp

```diff
@@ -19,13 +19,15 @@ VarValue PapyrusLeveledItem::GetNthForm(VarValue self,
   auto& loader = compatibilityPolicy->GetWorldState()->GetEspm();
   auto leveledItem = espm::Convert<espm::LVLI>(itemRecord.rec);
   if (leveledItem) {
-    auto vec =
-      LeveledListUtils::EvaluateList(loader.GetBrowser(), itemRecord, 1);
+    auto data = leveledItem->GetData(
+      compatibilityPolicy->GetWorldState()->GetEspmCache());
     int index = static_cast<int>(arguments[0]);
-    if (vec.size() > index) {
-      auto formId = itemRecord.ToGlobalId(vec[index].formId);
-      auto record = itemRecord.parent->LookupById(formId);
-      return VarValue(std::make_shared<EspmGameObject>(record));
+    if (static_cast<int>(data.numEntries) > index) {
```

Comment:
It would probably make more sense in casting index to size_t instead to ensure that there are no weird overflow bugs